.github/actions/do_build_icd/action.yml

```diff
@@ -0,0 +1,72 @@
+name: pull and build opencl icd loader
+description: pull icd loader and build with a particular toolchain, uploading opencl header and icd artefacts
+
+inputs:
+  target:
+    description: 'target architecture'
+
+runs:
+  using: "composite"
+  steps:
+    - name: calc vars
+      id: calc_vars
+      uses: ./.github/actions/calc_vars
+      with:
+        target: ${{ inputs.target }}  
+
+    - name: Install Ninja
+      uses: llvm/actions/install-ninja@main
```

Comment:
Despite us putting a lot of trust in LLVM already, this counts as a third-party action which should use a commit hash rather than a branch, I think. Likewise in do_build_tornado

Reply:
ah yes quite right, I thought I copied from somewhere I'll check.